via/static/scripts/video_player/components/VideoPlayerApp.tsx

```diff
@@ -382,24 +428,36 @@ export default function VideoPlayerApp({
               'flex flex-col'
             )}
           >
-            <Transcript
-              autoScroll={autoScroll}
-              transcript={transcript}
-              controlsRef={transcriptControls}
-              currentTime={timestamp}
-              filter={trimmedFilter}
-              onSelectSegment={segment => setTimestamp(segment.start)}
-            >
-              <div
-                data-testid="bucket-bar-channel"
-                className={classnames(
-                  // Provide a backdrop for bucket bar buttons. 20px of this
-                  // is overlaid by the sidebar's semi-transparent bucket
-                  // channel.
-                  'bg-gradient-to-r from-white to-grey-1 border-l w-[40px]'
-                )}
-              />
-            </Transcript>
+            {isLoading && (
+              <div className="flex justify-center p-8">
+                <Spinner data-testid="transcript-loading-spinner" size="md" />
+              </div>
+            )}
+            {isTranscript(transcript) && (
+              <Transcript
+                autoScroll={autoScroll}
+                transcript={transcript}
+                controlsRef={transcriptControls}
+                currentTime={timestamp}
+                filter={trimmedFilter}
+                onSelectSegment={segment => setTimestamp(segment.start)}
+              >
+                <div
+                  data-testid="bucket-bar-channel"
+                  className={classnames(
+                    // Provide a backdrop for bucket bar buttons. 20px of this
+                    // is overlaid by the sidebar's semi-transparent bucket
+                    // channel.
+                    'bg-gradient-to-r from-white to-grey-1 border-l w-[40px]'
+                  )}
+                />
+              </Transcript>
+            )}
+            {transcript instanceof Error && (
+              <div data-testid="transcript-error">
+                Unable to load transcript: {transcript.message}
+              </div>
+            )}
```

Comment:
This component is starting to get a little on the large side. We should start to think about factoring out some smaller pieces soonish.